### ISSUE TITLE
Support for async-safety through asgiref's Local

### ISSUE THREAD

**Issue by aprams** [citusdata/django-multitenant#176]
We are looking to switch to ASGI soon and are freshly using django-multitenant as well. As the `threading.local` is being used by django-multitenant, I'm afraid of cross-request tenant leakages. 
I therefore suggest an [approach similar to django-simple-history](https://github.com/jazzband/django-simple-history/blob/e368d839c981c60376cbffcdaba245b89766bff0/simple_history/models.py#LL43C1-L46C48): 

```
try:
    from asgiref.local import Local as LocalContext
except ImportError:
    from threading import local as LocalContext
```

This should use asgiref's `Local` as an asgi-safe alternative to `threading.locals` as per my understanding. As asgiref.local is meant as a drop-in replacement, the change should be straightforward.

**Comment by ArpanKIIT2017** [comment]
Yes, even I wanted to understand if django-multitenant is compatible with ASGI and Channels server. I am new to Channels and ASGI, so need some advice before I port my current Django app which uses this library into an ASGI mode for some of the new features.

**Comment by ArpanKIIT2017** [comment]
@darwing1210 Do we have an eta when this can be released?

**Comment by darwing1210** [comment]
> @darwing1210 Do we have an eta when this can be released?

I'm not sure, as it was never reviewed by the repo owners

**Comment by gurkanindibay** [comment]
Hey @darwing1210 @arpanpreneur I'm reviewing right now. Thanks for your contributions

**Comment by aprams** [comment]
@gurkanindibay This can be closed as per https://github.com/citusdata/django-multitenant/commit/fd84aeee9f71f66775a5f4256102f9010418a940 right?
